src/Avalonia.Base/Controls/INotSharedDeferredContent.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace Avalonia.Controls;
+
+/// <summary>
+/// Represents a not shared deferred content.
+/// </summary>
+public interface INotSharedDeferredContent: IDeferredContent
```

Comment:
While we could use a marker interface internally for the `ResourceDictionary` to use, I don't think it should be public here. Ideally it should only be known by the `ResourceDictionary`. We can then simplify things by reusing the existing `IDeferredContent` implementations.

See other comments for details.

Reply:
Now that's it's not used at all in public API, I think you can completely remove it and simply check for `NotSharedDeferredItem` in the `ResourceDictionary` instead.